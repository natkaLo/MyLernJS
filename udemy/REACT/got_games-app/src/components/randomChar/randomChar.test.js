import React from "react";
import RandomChar from './randomChar';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

//тест сьют
describe('Testing<RandomChar/>', () => {
    const char = shallow(<RandomChar/>); //создаем копию с помощью enzyme
   describe('Testing snap and state', () => {
            //тест кейс. Начинается с ключевого слова it
        it('RandomChar have rendered correctly', () => {
            //создаем компонент который надо протестировать. Создаем с помощью renderer из библиотеки react-test-renderer
            const char = renderer.create(<RandomChar/>).toJSON(); //получаем джейсон объект
            //хотим создать снимок как этот компонент должен выглядеть
            expect(char).toMatchSnapshot();
            //запуск теста - в терминале
            //npm run test

        });
        // it('should render a title', () => {
        //     const wrapper = shallow(<Document title = 'Some title'/>);
        //     expect(wrapper.prop('title')).toEqual('Some title');
        // })
       
        it('RandomChar  have rendered correctly', () => {
            expect(char).toMatchSnapshot();
        });
        // it('RandomChar state "char" is empty object', () => {
        //     expect(char.state().char).toBeObject();//toBeObject из Jest extended
        // });

        it('RandomChar state "loading" is true', () => {
            expect(char.state().loading).toBeTruthy();
        });
        it('RandomChar state "error" is false', () => {
            expect(char.state().error).toBeFalsy();
        });
   });
   //проверяем работу функций
   describe('Handlers tests', () =>{
        it('testing onCharLoade', () =>{
            char.instance().onCharLoaded(); //вызываем функцию
            expect(char.state().loading).toBeFalsy();//проверяем результат
        });
        it('testing onError', () =>{
            char.instance().onError(); //вызываем функцию
            expect(char.state().loading).toBeFalsy();//проверяем результат
            expect(char.state().error).toBeTruthy();//проверяем результат
        });
        it('testing updateChar', () =>{
            char.instance().updateChar(); //вызываем функцию
            expect(char.state().loading).toBeFalsy();//проверяем результат
           
        });
   });
});