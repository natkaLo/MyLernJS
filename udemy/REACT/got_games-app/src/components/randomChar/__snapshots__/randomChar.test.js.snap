// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing<RandomChar/> RandomChar  have rendered correctly 1`] = `ShallowWrapper {}`;

exports[`Testing<RandomChar/> RandomChar have rendered correctly 1`] = `
<div
  className="random-block rounded"
>
  <div
    className="loadingio-spinner-spin-91jgnvmw819"
  >
    <div
      className="ldio-6yxzrqei8yb"
    >
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing<RandomChar/> RandomChar state "char" is empty object 1`] = `ShallowWrapper {}`;

exports[`Testing<RandomChar/> Testing snap and state RandomChar  have rendered correctly 1`] = `ShallowWrapper {}`;

exports[`Testing<RandomChar/> Testing snap and state RandomChar have rendered correctly 1`] = `
<div
  className="random-block rounded"
>
  <div
    className="loadingio-spinner-spin-91jgnvmw819"
  >
    <div
      className="ldio-6yxzrqei8yb"
    >
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
